include/SugarFolders/SugarFolders.php

```diff
@@ -561,17 +561,17 @@ public function retrieveFoldersForProcessing($user, $subscribed = true)
 
         $found = array();
         while ($a = $this->db->fetchByAssoc($r)) {
-            if ((($a['folder_type'] == $myEmailTypeString) ||
-                    ($a['folder_type'] == $myDraftsTypeString) ||
-                    ($a['folder_type'] == $mySentEmailTypeString)) &&
-                ($a['created_by'] != $current_user->id)
-            ) {
+            if($a['folder_type'] == $myEmailTypeString) {
+                if (!isset($found[$a['id']])) {
+                    $found[$a['id']] = true;
 
-                continue;
-            } // if
-            if (!isset($found[$a['id']])) {
-                $found[$a['id']] = true;
-                $return[] = $a;
+                    $children = $this->db->query('SELECT * FROM folders WHERE parent_folder = "'. $a['id'] . '"');
+                    while ($b = $this->db->fetchByAssoc($children)) {
+                        $a['children'][] = $b;
+                    }
+
+                    $return[] = $a;
```

Comment:
Unhandled negative-case (empty) result. What happening when I have not set any folder to show?